core/server/api_container/server/startosis_engine/startosis_runner.go

```diff
@@ -28,15 +32,29 @@ func NewStartosisRunner(interpreter *StartosisInterpreter, validator *StartosisV
 		startosisInterpreter: interpreter,
 		startosisValidator:   validator,
 		startosisExecutor:    executor,
+		mutex:                &sync.Mutex{},
 	}
 }
 
 func (runner *StartosisRunner) Run(ctx context.Context, dryRun bool, parallelism int, packageId string, serializedStartosis string, serializedParams string) <-chan *kurtosis_core_rpc_api_bindings.StarlarkRunResponseLine {
+	runner.mutex.Lock()
```

Comment:
the defer to unlock this should be right here on line 41

Reply:
Though it does not matter; because we only run go-routine per invocation; but yeah your way is more verbose. will do that.